gstreamer-rtsp-server/src/auto/r_t_s_p_stream.rs

```diff
@@ -0,0 +1,591 @@
+
+    fn get_caps(&self) -> Option<gst::Caps>;
+
+    fn get_control(&self) -> Option<String>;
+
+    fn get_current_seqnum(&self) -> u16;
+
+    fn get_dscp_qos(&self) -> i32;
+
+    fn get_index(&self) -> u32;
+
+    fn get_joined_bin(&self) -> Option<gst::Bin>;
+
+    fn get_mtu(&self) -> u32;
+
+    fn get_multicast_address(&self, family: gio::SocketFamily) -> Option<RTSPAddress>;
+
+    fn get_multicast_iface(&self) -> Option<String>;
+
+    fn get_profiles(&self) -> gst_rtsp::RTSPProfile;
+
+    fn get_protocols(&self) -> gst_rtsp::RTSPLowerTrans;
+
+    fn get_pt(&self) -> u32;
+
+    fn get_publish_clock_mode(&self) -> RTSPPublishClockMode;
+
+    fn get_retransmission_pt(&self) -> u32;
+
+    fn get_retransmission_time(&self) -> gst::ClockTime;
+
+    fn get_rtcp_socket(&self, family: gio::SocketFamily) -> Option<gio::Socket>;
+
+    fn get_rtp_socket(&self, family: gio::SocketFamily) -> Option<gio::Socket>;
+
+    fn get_rtpinfo(&self) -> Option<(u32, u32, u32, gst::ClockTime)>;
+
+    fn get_rtpsession(&self) -> Option<glib::Object>;
+
+    //fn get_server_port(&self, server_port: /*Ignored*/gst_rtsp::RTSPRange, family: gio::SocketFamily);
+
+    fn get_sinkpad(&self) -> Option<gst::Pad>;
+
+    fn get_srcpad(&self) -> Option<gst::Pad>;
+
+    fn get_srtp_encoder(&self) -> Option<gst::Element>;
+
+    fn get_ssrc(&self) -> u32;
+
+    fn has_control<'a, P: Into<Option<&'a str>>>(&self, control: P) -> bool;
+
+    fn is_blocking(&self) -> bool;
+
+    fn is_client_side(&self) -> bool;
+
+    //fn is_transport_supported(&self, transport: /*Ignored*/&mut gst_rtsp::RTSPTransport) -> bool;
+
+    fn join_bin<P: IsA<gst::Bin>, Q: IsA<gst::Element>>(&self, bin: &P, rtpbin: &Q, state: gst::State) -> Result<(), glib::error::BoolError>;
+
+    fn leave_bin<P: IsA<gst::Bin>, Q: IsA<gst::Element>>(&self, bin: &P, rtpbin: &Q) -> Result<(), glib::error::BoolError>;
+
+    fn recv_rtcp(&self, buffer: &gst::Buffer) -> gst::FlowReturn;
+
+    fn recv_rtp(&self, buffer: &gst::Buffer) -> gst::FlowReturn;
+
+    fn remove_transport(&self, trans: &RTSPStreamTransport) -> Result<(), glib::error::BoolError>;
+
+    fn request_aux_sender(&self, sessid: u32) -> Option<gst::Element>;
+
+    fn reserve_address(&self, address: &str, port: u32, n_ports: u32, ttl: u32) -> Option<RTSPAddress>;
+
+    fn set_address_pool<'a, P: Into<Option<&'a RTSPAddressPool>>>(&self, pool: P);
+
+    fn set_blocked(&self, blocked: bool) -> bool;
+
+    fn set_buffer_size(&self, size: u32);
+
+    fn set_client_side(&self, client_side: bool);
+
+    fn set_control<'a, P: Into<Option<&'a str>>>(&self, control: P);
+
+    fn set_dscp_qos(&self, dscp_qos: i32);
+
+    fn set_mtu(&self, mtu: u32);
+
+    fn set_multicast_iface<'a, P: Into<Option<&'a str>>>(&self, multicast_iface: P);
+
+    fn set_profiles(&self, profiles: gst_rtsp::RTSPProfile);
+
+    fn set_protocols(&self, protocols: gst_rtsp::RTSPLowerTrans);
+
+    fn set_pt_map(&self, pt: u32, caps: &gst::Caps);
```

Comment:
Nullable IIRC

Reply:
not really, passing NULL has no effect now:

```
c3ed1e69 (Mathieu Duponchelle    2018-02-10 00:07:25 +0100 2245)   if (!GST_IS_CAPS (caps))
c3ed1e69 (Mathieu Duponchelle    2018-02-10 00:07:25 +0100 2246)     return;
```

prior to that it would just fail at the subsequent call to `gst_caps_ref`